lib/Doctrine/Inflector/Inflector.php

```diff
@@ -78,6 +86,398 @@ public function capitalize(string $string, string $delimiters = " \n\t\r\0\x0B-"
         return ucwords($string, $delimiters);
     }
 
+    /**
+     * Checks if the given string seems like it has utf8 characters in it.
+     *
+     * @param string $string The string to check for utf8 characters in.
+     */
+    public function seemsUtf8(string $string) : bool
+    {
+        for ($i = 0; $i < strlen($string); $i++) {
+            if (ord($string[$i]) < 0x80) {
+                continue; // 0bbbbbbb
+            } elseif ((ord($string[$i]) & 0xE0) === 0xC0) {
+                $n = 1; // 110bbbbb
+            } elseif ((ord($string[$i]) & 0xF0) === 0xE0) {
+                $n = 2; // 1110bbbb
+            } elseif ((ord($string[$i]) & 0xF8) === 0xF0) {
+                $n = 3; // 11110bbb
+            } elseif ((ord($string[$i]) & 0xFC) === 0xF8) {
+                $n = 4; // 111110bb
+            } elseif ((ord($string[$i]) & 0xFE) === 0xFC) {
+                $n = 5; // 1111110b
+            } else {
+                return false; // Does not match any model
+            }
+
+            for ($j = 0; $j < $n; $j++) { // n bytes matching 10bbbbbb follow ?
+                if (++$i === strlen($string) || ((ord($string[$i]) & 0xC0) !== 0x80)) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Remove any illegal characters, accents, etc.
+     *
+     * @param  string $string String to unaccent
+     *
+     * @return string Unaccented string
+     */
+    public function unaccent(string $string) : string
+    {
+        if (preg_match('/[\x80-\xff]/', $string) === false) {
+            return $string;
+        }
+
+        if ($this->seemsUtf8($string)) {
+            $characters = [
```

Comment:
Can we move all these into a private constants?

Reply:
The whole array or just 1 constant per character?

Reply:
Whole arrays.

Reply:
I don't think you can do that because of the `chr()` call:

```php
    private const TEST = [
        chr(195) . chr(128) => 'A',
    ];
```

```
PHP Fatal error:  Constant expression contains invalid operations in /data/projects/inflector/lib/Doctrine/Inflector/Inflector.php on line 23
```

Reply:
True but you could just convert them to strings a la "\xFF\xFF" style.

Reply:
w00f. If someone is up for making that change feel free :)

Reply:
Nevermind, someone else can fix it. Here you go. https://hastebin.com/peviqubanu.php

Reply:
There is no need to use chr(), just like @garrettw says.

Reply:
Sweet. I'll plug that in and make the change.